File: app/src/main/java/com/games/memorygame/ui/board/BoardMvpView.java
package com.games.memorygame.ui.board;


import com.games.memorygame.model.Photo;
import com.games.memorygame.model.game.Player;
import com.games.memorygame.ui.MvpView;

import java.util.List;

public interface BoardMvpView extends MvpView {

    void showDashboard(List<Photo> photos, int rows, int columns);
    void onError(String errorMessage);
    void onNetworkError();
    void updateScore(List<Player> players);
    void updatePlayerTurn(List<Player> players);
    void setUpTimer(int seconds);
}

File: app/src/main/java/com/games/memorygame/ui/board/BoardMvpView.kt
package com.games.memorygame.ui.board


import com.games.memorygame.model.Photo
import com.games.memorygame.model.game.Player
import com.games.memorygame.ui.MvpView

interface BoardMvpView : MvpView {

    fun showDashboard(photos: List<Photo>, rows: Int, columns: Int)
    fun onError(errorMessage: String)
    fun onNetworkError()
    fun updateScore(players: List<Player>)
    fun updatePlayerTurn(players: List<Player>)
    fun setUpTimer(seconds: Int)
}
